### PR TITLE
[cli] Add tests to compare dev/prd image optimization

### PR DESCRIPTION
**Pull request by styfle**
Add E2E test for to test image optimization against `vc dev` as well as a prod deployment.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR (CH13104)

### CODE REVIEW

**Review by juancampa — COMMENTED** [review]
@styfle does `vc dev` return `last-modified`? That's something that people have asked about, imgix returns it and we simply forward it

**Comment by styfle** [comment]
@juancampa No it does not. The `next dev` implementation uses `etag` instead (per @Timer) which is why it was omitted from these tests. See https://github.com/vercel/next.js/pull/18986

**Comment by juancampa** [comment]
Got it, that's an inconsistency that we might want to fix later

**Review by juancampa — DISMISSED** [review]

**Comment by Timer** [comment]
@juancampa I think the inconsistency is inherent to the fact that Next.js is optimizing the image itself (thus has the Etag readily at its disposal) verses on Vercel where we proxy Imgix. I don't see the need to be consistent here.

**Comment by juancampa** [comment]
I could see users being confused by etags appearing in their `vc dev` experience, but not in prod. But yeah, for now I think it's okay to accept that there are some (non fundamental) differences between the two environments

**Review by juancampa — APPROVED** [review]

**Review by Timer — APPROVED** [review]

File: packages/now-cli/test/dev/fixtures/30-next-image-optimization/vercel.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "version": 2,
+  "build": {
+    "env": {
+      "FORCE_BUILDER_TAG": "canary"
```
Comment (TooTallNate) [inline packages/now-cli/test/dev/fixtures/30-next-image-optimization/vercel.json:5]: Why is this necessary?

**Review by TooTallNate — APPROVED** [review]

File: packages/now-cli/test/dev/fixtures/30-next-image-optimization/vercel.json (hunk 1 above)
Comment (styfle) [reply packages/now-cli/test/dev/fixtures/30-next-image-optimization/vercel.json:5]: Its not necessary but it will help us catch differences in `@vercel/next` canary before they go to stable
